src/ui/public/vis/request_handlers/courier.js

```diff
@@ -55,11 +43,15 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
         return timefilter.get(searchSource.index(), timeRange);
       });
 
-      removeSearchSourceParentTimefilter(requestSearchSource);
-
-      if (queryFilter && vis.editorMode) {
-        searchSource.set('filter', queryFilter.getFilters());
-        searchSource.set('query', appState.query);
+      // in editor mode we want to set filter and query on searchSource so they get
+      // stored with visualization, however on dashboard we set them on requestSearchSource
+      // so we don't override the stored ones
+      if (vis.editorMode) {
```

Comment:
I think it's okay for now, but we should still figure out, if we can find a better way than that.

Reply:
I feel like the editor just shouldn't pass in the filters or query to `<visualize>`, instead the editor will set them on the vis `searchSource` and `<visualize>` will default to no extra filters or queries.